spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -521,10 +531,7 @@ function chanUpgradeTry(
             currentChannel.upgradeSequence = counterpartyUpgradeSequence - 1
         }
 
-        initUpgradeChannelHandshake(portIdentifier, channelIdentifier, upgradeFields, counterpartyUpgrade.timeout)
-    } else if currentChannel.state == INITUPGRADE {
-        existingUpgrade = publicStore.get(channelUpgradePath)
-        abortTransactionUnless(existingUpgrade.fields == upgradeFields)
+        initUpgradeChannelHandshake(portIdentifier, channelIdentifier, upgradeFields)
     }
```

Comment:
as noted offline, in the crossing hellos case, we likely need to handle validating relayer provided connection hops since we restore when existing upgrade fields != proposed upgrade fields

As noted offline, my recommendation would be to use the existing upgrade if it exists